clusterman/kubernetes/kubernetes_cluster_connector.py

```diff
@@ -241,6 +241,10 @@ def _count_batch_tasks(self, node_ip: str) -> int:
                     break
         return count
 
+    def _freeze_agent(self, agent_id: str) -> None:
+        body = {"spec": {"unschedulable": True}}
```

Comment:
how does this compare to using a `NoSchedule` taint? if we set a `NoSchedule` taint with a descriptive name, I think it'll make it easier to determine why a node is unschedulable?

Reply:
in that case, can we just create the taint directly (with a more descriptive name than unschedulable)?

Reply:
Sure, But how can we achieve stopping new pods scheduling ?
I "cordon" node when we mark node for termination. it will avoid new pods scheduling before real termination.
The gap between "marking for termination" and "real termination" is more than 30 seconds for jenkins pool. 
And kube-scheduler schedules new pods (jenkins job) in this gap, and we delete pods right before real termination (node). deletion of these pods causes issue for someone's pipeline.

Reply:
`cordon` does is set `node.Spec.Unschedulable` to be true (https://github.com/kubernetes/kubectl/blob/master/pkg/drain/cordon.go#L89). as you say, this stops pods from being scheduled on a host.
when launching a new pod, the scheduler also checks whether pods have tolerations for all the taints on the host. taints can have two effects - `NoSchedule` or `NoExecute`. both have the impact of preventing any pods being scheduled unless they have a matching toleration, but NoExecute also removes running pods.

So they both have the same effect for us - stopping pods from being scheduled on a host (though, if we use a taint, an operator *could* circumvent it by setting the appropriate taint on a pod. unlikely to be a problem for us).

the reason I'm advocating for a taint instead of just marking a node as `unschedulable` is that the taint gives us the opportunity to put a descriptive reason on *why* we're stopping pods being launched. a node might be unschedulable for a bunch of reasons - it'd be good if we could avoid adding another reason (clusterman is shutting the node down) into that mix.

Reply:
I got your point! it is good idea. I will taint node with something like "ClustermanScaleDown=true:NoSchedule". Actually I will try to find better name :) thank you!

Reply:
Sure, Added, thanks!